js/Source/Charts/Chart.js

```diff
@@ -236,6 +236,22 @@ Chart = new Class({
     this.chart.setSize(width, height, false);
     // redraw the backgrounds we added so they fit the new dimentions
     this._drawBackgrounds();
+
+
+    // resize the legend, which requires us to overwrite
+    // numerous width values calculated when the chart was
+    // initialized, recalculate them, then reset the options
+    // of the chart to the new sizies, destroy the legend items
+    // and re-render it
+    delete this.options.legendItemWidth;
+    delete this.options.legendWidth;
+    this._setV2Defaults();
+    this.chart.legend.options.width = this.options.legendWidth;
+    this.chart.legend.itemStyle.width = this.options.legendItemWidth;
+    this.chart.legend.options.itemWidth = this.options.legendItemWidth;
+    this.chart.legend.maxItemWidth = this.options.legendItemWidth;
+    this.chart.series.each(this.chart.legend.destroyItem, this);
+    this.chart.legend.render();
```

Comment:
nah. It's instantaneous. It basically destroys some DOM elements and recreates them.